homeassistant/components/hassio/__init__.py

```diff
@@ -590,7 +590,7 @@ async def async_handle_core_service(call: ServiceCall) -> None:
     await async_setup_addon_panel(hass, hassio)
 
     # Setup hardware integration for the detected board type
-    async def _async_setup_hardware_integration(hass):
+    async def _async_setup_hardware_integration(_: datetime) -> None:
```

Comment:
If the first/original `get_os_info` attempt succeeded, then `async_track_point_in_utc_time` was never called.
However, in my testing, if the first `get_os_info` attempt failed, then the retry mechanism would fail.